frontend/controller/actions/index.js

```diff
@@ -191,8 +187,9 @@ sbp('sbp/selectors/register', {
         contractID,
         contractName,
         data: [
-          ...keyShares.map((data) => ['chelonia/out/keyShare', { data: encryptedOutgoingData(contractID, CEKid, data) }]),
-          ['chelonia/out/keyUpdate', { data: updatedKeys }]
+          ...(keyShares[0] ?? []),
```

Comment:
Function updated to support arbitrary operations. This also makes this function generic and no longer GI-specific.